src/main/scala/cromwell/binding/Scope.scala

```diff
@@ -26,33 +27,50 @@ object Scope {
     def getChildrenList(ast: Ast): Seq[Ast] = Option(ast.getAttribute("body")) collect {
       case list: AstList => list.asScala.toVector
     } getOrElse Seq.empty collect {
-      case ast: Ast => ast
+      case node: Ast
+        if node.getName == AstNodeName.Call ||
+          node.getName == AstNodeName.Scatter =>
+        node
+    }
+
+    /**
+     * Sets the child scopes on this parent from the ast.
+     * @param parentAst Parent ast.
+     * @param parentScope Parent scope.
+     */
+    def setChildScopes(parentAst: Ast, parentScope: Scope): Unit = {
+      //Increment index if needed
+      scopeIndexes.put(parentScope.getClass, scopeIndexes.getOrElse(parentScope.getClass, 0) + 1)
+      parentScope.children = for {
+        child <- getChildrenList(parentAst)
+        scope = generateScopeTree(child, namespaces, tasks, wdlSyntaxErrorFormatter, parentScope)
+      } yield scope
     }
 
     /**
      * Generate a scope from the ast parameter and all its descendants recursively.
-     * @return Some(instance of Scope) if the node has a Scope equivalent, None otherwise
+     * @return Scope equivalent
      */
-    def generateScopeTree(node: Ast, namespaces: Seq[WdlNamespace], tasks: Seq[Task], wdlSyntaxErrorFormatter: WdlSyntaxErrorFormatter, parent: Option[Scope]): Option[Scope] = {
-      val scope: Option[Scope] = node match {
-        case w: Ast if w.getName == AstNodeName.Workflow => Option(Workflow(w, wdlSyntaxErrorFormatter, parent))
-        case c: Ast if c.getName == AstNodeName.Call => Option(Call(c, namespaces, tasks, wdlSyntaxErrorFormatter, parent))
-        case s: Ast if s.getName == AstNodeName.Scatter => Option(Scatter(s, scopeIndexes.getOrElse(classOf[Scatter], 0), parent))
-
-        /* Cases need to be added here to support new scopes */
-        case _ => None
+    def generateScopeTree(node: Ast,
+                          namespaces: Seq[WdlNamespace],
+                          tasks: Seq[Task],
+                          wdlSyntaxErrorFormatter: WdlSyntaxErrorFormatter,
+                          parent: Scope): Scope = {
+
+      val scope: Scope = node.getName match {
+        case AstNodeName.Call =>
+          Call(node, namespaces, tasks, wdlSyntaxErrorFormatter, Option(parent))
+        case AstNodeName.Scatter =>
+          Scatter(node, scopeIndexes.getOrElse(classOf[Scatter], 0), Option(parent))
```

Comment:
I guess we haven't really determined any syntax errors that can happen in a Scatter block creation... maybe when we get the "type evaluator" we can check to make sure that the expression in the block will evaluate to an `Array`